Library/ConnectionStringParser.ts

```diff
@@ -0,0 +1,45 @@
+import { ConnectionStringFields } from "../Declarations/Contracts";
+import Constants = require("../Declarations/Constants");
+
+class ConnectionStringParser {
+    private static _FIELDS_SEPARATOR = ";";
+    private static _FIELD_KEY_VALUE_SEPARATOR = "=";
+
+    public static parse(connectionString?: string): ConnectionStringFields {
+        if (!connectionString) {
+            return {};
+        }
+
+        const kvPairs = connectionString.split(ConnectionStringParser._FIELDS_SEPARATOR);
+
+        const result: ConnectionStringFields = kvPairs.reduce((fields: ConnectionStringFields, kv: string) => {
+            const kvParts = kv.split(ConnectionStringParser._FIELD_KEY_VALUE_SEPARATOR);
+
+            if (kvParts.length === 2) { // only save fields with valid formats
+                const key = kvParts[0].toLowerCase();
+                const value = kvParts[1];
+                fields[key] = value;
+            }
+            return fields;
+        }, {});
+
+        if (Object.keys(result).length > 0) {
+            // this is a valid connection string, so parse the results
+
+            if (result.endpointsuffix) {
+                // use endpoint suffix where overrides are not provided
+                const locationPrefix = result.location ? result.location + "." : "";
```

Comment:
Location can be pre-pended even if endpoints suffix is not provided.

Reply:
Can you clarify what you mean?

Reply:
Please do not delete my comments.

We have an email thread where we've defined how we intend to handle Locations. Locations are to only apply to the Endpoint Suffix. 
If you have an issue with this please contact me offline.